### DISCUSSION TITLE
Increased support for non-centred parameterisations

### DISCUSSION THREAD

**Discussion by drbenvincent** [pymc-devs/pymc3#6703] [Ideas]
There’s been a small amount of discussion about this, but seeking broader input on this idea... The proposal is to increase support for non-centered parameterisations in PyMC (i.e. for hierarchical models). I think there are 3 possible approaches, but feel free to suggest more.

**Idea 1 - Create a `pm.NormalNonCentered` distribution**
Something like 
```python
def NormalNonCentered(varname: str, mu, sigma, dims: str):
    _mu = pm.Normal(f"z_{varname}", dims=dims)
    return pm.Deterministic(varname, mu + _mu * sigma, dims=dims)
```

**Idea 2 - Add a `parameterisation` kwarg for `pm.Normal`**

**Idea 3 - Add it as a transformation, eg `pymc.distributions.transforms.NonCentered`**

**Comment by theorashid** [comment]
I'd rather see idea 2 or 3. Idea 2 is probably best from an end user point of view, but probably bloats the codebase. I've used numpyro's non-centred transformation in the past and it's very easy to use (see [here](https://github.com/pyro-ppl/numpyro/blob/master/examples/mortality.py#L120), the tranformed variables are in the block above using `LocScaleReparam`) – I don't know the internals though. Pyro has many more transforms implemented than numpyro

**Comment by ferrine** [comment]
Can it be on class structure level? E.g with a LocScale base class?

**Reply by theorashid** [reply]
Do you mean like numpyro's `LocScaleReparam(Reparam)` [class](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/infer/reparam.py#L63)?
